helix-view/src/document.rs

```diff
@@ -139,14 +143,59 @@ pub fn canonicalize_path(path: &Path) -> std::io::Result<PathBuf> {
 use helix_lsp::lsp;
 use url::Url;
 
+// Temporary
+#[inline(always)]
+fn checked_add(num: usize, opt: Option<usize>) -> Option<usize> {
+    if let Some(n) = opt {
+        n.checked_add(num)
+    } else {
+        None
+    }
+}
+
+// Temporary
+#[inline(always)]
+fn checked_next_power_of_two(opt: Option<usize>) -> Option<usize> {
+    opt.map(|n| n.next_power_of_two())
+}
+
+// Temporary
+#[inline(always)]
+fn checked_min(one: Option<usize>, other: Option<usize>) -> Option<usize> {
+    if let Some(a) = one {
+        if let Some(b) = other {
+            Some(::core::cmp::min(a, b))
+        } else {
+            Some(a)
+        }
+    } else {
+        other
+    }
+}
+
 impl Document {
-    pub fn new(text: Rope) -> Self {
+    pub fn new() -> Self {
+        let text = Rope::from("\n");
+
+        Self::from(text, None, None)
+    }
+
+    pub fn from(
+        text: Rope,
+        encoding: Option<&'static encoding_rs::Encoding>,
+        file_encoding: Option<&'static encoding_rs::Encoding>,
+    ) -> Self {
         let changes = ChangeSet::new(&text);
         let old_state = None;
+        let encoding =
```

Comment:
Use the static declaration: https://docs.rs/encoding_rs/0.8.28/encoding_rs/static.UTF_8.html